src/wazuh_modules/vulnerability_scanner/src/databaseFeedManager/eventDecoder.hpp

```diff
@@ -17,81 +17,142 @@
 #include "cve5_schema.h"
 #include "eventContext.hpp"
 #include "flatbuffers/idl.h"
+#include "packageTranslation_generated.h"
+#include "packageTranslation_schema.h"
+#include "stringHelper.h"
 
 /**
  * @brief EventDecoder class.
  *
  */
 class EventDecoder final : public AbstractHandler<std::shared_ptr<EventContext>>
 {
-public:
+private:
+    const std::unique_ptr<Utils::RocksDBWrapper>& m_translationsDatabase;
+
     /**
-     * @brief Handles request and passes control to the next step of the chain.
+     * @brief Process a CVE5 or Translation message.
      *
-     * @param data Scan context.
-     * @return std::shared_ptr<ScanContext> Abstract handler.
+     * @param data Event context.
      */
-    std::shared_ptr<EventContext> handleRequest(std::shared_ptr<EventContext> data) override
+    void processEvent(std::shared_ptr<EventContext> data)
     {
-        if ("create" == data->resource.at("type"))
+        if (data->resource.contains("resource"))
         {
-            flatbuffers::Parser parser;
-
-            if (!parser.Parse(cve5_SCHEMA) || !parser.Parse(data->resource.at("payload").dump().c_str()))
+            if (!Utils::startsWith(data->resource.at("resource"), "CVE-") &&
```

Comment:
Performance note: we are executing this method 3 times in the class, making `isCVEResource` a member variable could avoid this.